File: Python/BOJ/1541 잃어버린괄호.py
import sys
S = sys.stdin.readline().strip().split('-') # - 기준으로 괄호를 치면 최소로 만들 수 있음
temp = []
for i in S: # 각 요소에 대해
    cnt = 0
    for j in i.split('+'): # +로 짼 요소들을
        cnt += int(j) # 더하고
    temp.append(cnt) # 모아주자
# 모아준 temp들을 맨 앞 인덱스 제외하고 전부 빼면 최소값
result = temp[0]
for i in temp[1:]:
    result -= i
print(result)

File: Java/BOJ/1541 잃어버린 괄호.java
import java.io.*;
import java.util.*;

public class Main {    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] S = br.readLine().split("-");
        int[] x = new int[S.length];
        
        for (int i = 0; i < S.length; i++) {
        	String[] y = S[i].split("\\+");
        	int cnt = 0;
        	for (String j : y ) {
        		cnt += Integer.parseInt(j);
        	}
        	x[i] = cnt;
        }
        
        int result = x[0];
        for (int i = 1 ; i < x.length; i++) {
        	result -= x[i];
        }
        System.out.println(result);
    }
}